web/src/emoji.ts

```diff
@@ -1,34 +1,93 @@
 import _ from "lodash";
 
+type RealmEmoji = Record<
+    string,
+    {
+        id: number;
+        author_id: number;
+        deactivated: boolean;
+        name: string;
+        source_url: string;
+        still_url: string | null;
+    }
+>;
+
+type EmojiCodes = {
```

Comment:
Let's call this type `ServerUnicodeEmojiData` -- it's the data the server provides about unicode emoji.